webpack/assets/javascripts/react_app/components/hosts/powerStatus/PowerStatus.js

```diff
@@ -1,11 +1,14 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { simpleLoader } from '../../../common/Loader';
-import './PowerStatusInner.scss';
+import { simpleLoader } from '../../common/Loader';
+import './PowerStatus.scss';
 
-const PowerStatusInner = ({ state, title, statusText, error }) => {
+const PowerStatus = ({ state, title, statusText, error }) => {
   if (error) {
-    const errorTitle = [title, statusText].join(' ').trim();
+    let errorTitle = error.message;
+    if (title || statusText) {
+      errorTitle = `${title} ${statusText}`.trim();
```

Comment:
how about moving this logic to selectors? this is a derived data from the store, this will trigger on each re-render.
with selectors, we can compute the wanted info from the state and make the component simpler:
```js
    <span
      className={`fa fa-power-off host-power-status ${classState}`}
      title={powerStatusTitle}
    />
```
while `classState` and `powerStatusTitle` are derived data  via selectors

Reply:
thanks, updated - I really like this change :+1: